wasm/src/main/java/com/dylibso/chicory/wasm/types/FunctionSection.java

```diff
@@ -1,17 +1,15 @@
 package com.dylibso.chicory.wasm.types;
 
-import java.util.Arrays;
-import java.util.Objects;
+import java.util.ArrayList;
 
 public class FunctionSection extends Section {
-    private int[] typeIndices;
-    private int count;
+    private final ArrayList<Integer> typeIndices;
```

Comment:
Using lists instead of arrays is a nice improvement.

Note that it's typically better to declare as the interface type, so `List<Integer>`, as the calling code doesn't (and shouldn't) depend on the concrete type.

Reply:
Yes, too much copy-pasting! Fixing it :slightly_smiling_face: